cli/src/main.rs

```diff
@@ -1,48 +1,33 @@
+#![allow(dead_code)]
+#![deny(warnings)]
```

Comment:
```suggestion
```
It's kinda very opinionated to always treat all warnings as errors. It slows the development process when trying to remove code that triggers unused imports, and new warnings may emerge in future versions of Rust. The no warning policy can always be enforced later, for instance in the CI.